lib/ansible/modules/files/unarchive.py

```diff
@@ -435,7 +435,7 @@ def is_unarchived(self):
             # DEBUG
 #            err += "%s%s %10d %s\n" % (ztype, permstr, size, path)
 
-            dest = os.path.join(self.dest, path)
+            dest = os.path.join(self.b_dest, to_bytes(path, errors='surrogate_or_strict'))
```

Comment:
Use `b_dest` as the var here.  It'll likely require more changes, but for long lived bytes, we should differentiate by using `b_` prefix